app/commands/v2/publish.rb

```diff
@@ -104,61 +115,50 @@ def valid_update_types
         %w(major minor republish links)
       end
 
-      def content_item
-        @content_item ||= ContentItem.find_by(
-          id: pessimistic_content_item_scope.pluck(:id),
-          state: "draft",
-        )
-      end
-
       def already_published?
-        ContentItem.exists?(content_id: content_id, locale: locale, state: "published")
-      end
-
-      def pessimistic_content_item_scope
-        ContentItem.where(content_id: content_id, locale: locale).lock
+        document.editions.exists?(state: "published")
       end
 
       def clear_published_items_of_same_locale_and_base_path
         SubstitutionHelper.clear!(
-          new_item_document_type: content_item.document_type,
-          new_item_content_id: content_item.content_id,
+          new_item_document_type: edition.document_type,
+          new_item_content_id: document.content_id,
           state: "published",
-          locale: content_item.locale,
-          base_path: content_item.base_path,
+          locale: document.locale,
+          base_path: edition.base_path,
           downstream: downstream,
           callbacks: callbacks,
           nested: true,
         )
       end
 
       def set_public_updated_at
-        return if content_item.public_updated_at.present?
+        return if edition.public_updated_at.present?
 
         if update_type == "major"
-          content_item.update_attributes!(public_updated_at: Time.zone.now)
+          edition.update_attributes!(public_updated_at: Time.zone.now)
         elsif update_type == "minor"
-          content_item.update_attributes!(public_updated_at: previous_item.public_updated_at)
+          edition.update_attributes!(public_updated_at: previous_item.public_updated_at)
         end
       end
 
       def set_first_published_at
-        return if content_item.first_published_at.present?
-        content_item.update_attributes!(first_published_at: Time.zone.now)
+        return if edition.first_published_at.present?
+        edition.update_attributes!(first_published_at: Time.zone.now)
       end
 
       def publish_redirect(previous_base_path, locale)
-        draft_redirect = ContentItem.find_by(
+        draft_redirect = Edition.joins(:document).find_by(
```

Comment:
maybe the default scope on `Edition` could include this join. i'm guessing it would more or less always be needed?